libs/payments/cart/src/lib/checkout.service.ts

```diff
@@ -701,4 +703,49 @@ export class CheckoutService {
 
     return upgradedSubscription;
   }
+
+  async determineCheckoutAmount({
+    eligibility,
+    customer,
+    priceId,
+    currency,
+    taxAddress,
+  }: {
+    eligibility: SubscriptionEligibilityResult;
+    priceId: string;
+    currency: string;
+    taxAddress: TaxAddress;
+    customer?: StripeCustomer;
+  }) {
+    eligibility.subscriptionEligibilityResult
+    if (eligibility.subscriptionEligibilityResult === EligibilityStatus.UPGRADE) {
+      assert(
+        'fromPrice' in eligibility,
+        'fromPrice not present for upgrade cart'
+      );
+      assert(customer, 'Customer is required for upgrade')
+      const fromSubscription =
+        await this.subscriptionManager.retrieveForCustomerAndPrice(
+          customer.id,
+          eligibility.fromPrice.id
+        );
+      assert(fromSubscription, 'Subscription required');
+      const fromSubscriptionItem = retrieveSubscriptionItem(fromSubscription);
+      const upcomingInvoice =
+        await this.invoiceManager.previewUpcomingForUpgrade({
+          priceId,
+          customer,
+          fromSubscriptionItem,
+        });
+      return upcomingInvoice.oneTimeChargeSubtotal;
+    } else {
+      const upcomingInvoice = await this.invoiceManager.previewUpcoming({
+        priceId,
+        currency,
+        customer,
+        taxAddress,
+      });
+      return upcomingInvoice.subtotal;
```

Comment:
nit: by having different returned values (`.subtotal` and `.oneTimeChargeSubtotal`) we open ourselves up to potential mismatched-value bugs. Can these be the same?